### DISCUSSION TITLE
What am I doing wrong?

### DISCUSSION THREAD

**Discussion by matthewpkemp** [YOURLS/YOURLS#3129] [Q&A] [answered]
Hey YOURLS community, it appears I've botched something.

My setup: YOURLS v 1.8.2 & Sleeky and it's only for personal use.

I thought everything was working well (I saw 301s in browser when testing my instance), but then I started noticing oddness w/ various sites hitting the logs w/ 200 and not 301s.  Going into curl I realized that the instance is returning:

```
< HTTP/2 200
< date: Tue, 30 Nov 2021 17:36:14 GMT
< server: Apache
< cache-control: max-age=600
< expires: Tue, 30 Nov 2021 17:46:14 GMT
< vary: Accept-Encoding,User-Agent
< x-xss-protection: 1; mode=block
< strict-transport-security: max-age=31536000; includeSubDomains
< x-content-type-options: nosniff
< content-type: text/html; charset=UTF-8
<
		<style>ul#admin_menu li:not(.frontend_link) {display: none}</style>		<script type="text/javascript">
		window.location="*REDACTED LONG URL*";
		</script>
```

I cleared browser cache and am seeing 200s again.  The plugins I have enabled are:

- 404 if not found
- Admin NoReCAPTCHA
- Public Prefix-N-Shorten
- Random ShortURLs
- Sleeky Backend

I suppose this is something silly somewhere, but I can't seem to see what setting I've munged.

Sample cases:
Cleared cache browser with shortened url - 200
CURL or other bot (ex twitterbot from logs) shortened url - 200
Logged in to admin interface with browser, then trying shortened url - 301

Was expecting that all of the above would be a 301.

**Comment by dgw** [comment, marked as answer]
Sounds like something is causing the headers to be sent before `yourls_redirect()` happens. If it's too late to modify the headers, YOURLS sets a JavaScript redirect like the one that can be seen in your snippet above.

I think this is a Sleeky bug. The `<style>` tag in your snippet appears verbatim [here](https://github.com/Flynntes/Sleeky/blob/f7978d9d426e4373a02785c5c008920f4bec7a96/sleeky-backend/plugin.php#L142-L147), and outputting that while plugins are loading would cause the observed behavior re: headers being sent already by the time YOURLS attempts a redirect—and also matches your sample cases, where the expected behavior occurs if you log in to YOURLS before trying a short link.

**Reply by dgw** [reply]
Dug a bit more into Sleeky's issue tracker, and it appears you're not alone. See e.g. Flynntes/Sleeky#77 and Flynntes/Sleeky#39.

**Reply by matthewpkemp** [reply]
Thanks for the quick turn around, you are correct.  Disabling fixes this.
